storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -607,6 +608,27 @@ protected String javaCmd(String cmd) {
         return ret;
     }
 
+    /**
+     * Extracting out to mock it for tests.
+     * @return true if on Linux.
+     */
+    protected boolean isOnLinux() {
+        return SystemUtils.IS_OS_LINUX;
+    }
+
+    private void prefixNumaPinningIfApplicable(String numaId, List<String> commandList) {
+        if (numaId != null) {
+            if (isOnLinux()) {
+                commandList.add("numactl");
+                commandList.add("--i=" + numaId);
```

Comment:
sorry to keep beating you on this :D  How about:
```
commandList.add(0, "numactl");
commandList.add(1, "--i=" + numaId);
```
since we want to add prefix to `commandList` no matter it's empty or not